src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdater.cs

```diff
@@ -148,17 +112,7 @@ private void EnqueueChecksumUpdate()
 
             private async Task UpdateSolutionChecksumAsync(CancellationToken cancellationToken)
             {
-                using (await _gate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
-                {
-                    // hold onto previous snapshot
-                    var previousSnapshot = _lastSnapshot;
-
-                    // create a new one (incrementally update the snapshot)
-                    _lastSnapshot = await _checksumService.CreateChecksumAsync(_service.Workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
-
-                    // let old one go.
-                    previousSnapshot?.Dispose();
-                }
+                await _service.Workspace.CurrentSolution.State.GetChecksumAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
i woudl prefer 'state' not be exposed, and instead people just call GetChecksymAsync on the solution itself.